client/src/main/java/io/split/service/SplitHttpClientKerberosImpl.java

```diff
@@ -0,0 +1,220 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.net.*;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class SplitHttpClientKerberosImpl implements SplitHttpClient {
+
+    private static final Logger _log = LoggerFactory.getLogger(SplitHttpClient.class);
+    private static final String HEADER_CACHE_CONTROL_NAME = "Cache-Control";
+    private static final String HEADER_CACHE_CONTROL_VALUE = "no-cache";
+    private static final String HEADER_API_KEY = "Authorization";
+    private static final String HEADER_CLIENT_KEY = "SplitSDKClientKey";
+    private static final String HEADER_CLIENT_MACHINE_NAME = "SplitSDKMachineName";
+    private static final String HEADER_CLIENT_MACHINE_IP = "SplitSDKMachineIP";
+    private static final String HEADER_CLIENT_VERSION = "SplitSDKVersion";
+
+    private final RequestDecorator _requestDecorator;
+    private final String _apikey;
+    private final SDKMetadata _metadata;
+
+    public static SplitHttpClientKerberosImpl create(RequestDecorator requestDecorator,
+                                                     String apikey,
+                                                     SDKMetadata metadata) throws URISyntaxException {
+        return new SplitHttpClientKerberosImpl(requestDecorator, apikey, metadata);
+    }
+
+    SplitHttpClientKerberosImpl(RequestDecorator requestDecorator,
+                                String apikey,
+                                SDKMetadata metadata) {
+        _requestDecorator = requestDecorator;
+        _apikey = apikey;
+        _metadata = metadata;
+    }
+
+    public synchronized SplitHttpResponse get(URI uri, FetchOptions options, Map<String, List<String>> additionalHeaders) {
+        HttpURLConnection getHttpURLConnection = null;
+        try {
+            getHttpURLConnection = (HttpURLConnection) uri.toURL().openConnection();
+            return _get(getHttpURLConnection, options, additionalHeaders);
+        } catch  (Exception e) {
+            throw new IllegalStateException(String.format("Problem in http get operation: %s", e), e);
+        } finally {
+            try {
+                if (getHttpURLConnection != null) {
+                    getHttpURLConnection.disconnect();
+                }
+            } catch (Exception e) {
+                _log.error(String.format("Could not close HTTP URL Connection: %s", e), e);
+            }
+        }
+    }
+    public SplitHttpResponse _get(HttpURLConnection getHttpURLConnection, FetchOptions options, Map<String, List<String>> additionalHeaders) {
+        InputStreamReader inputStreamReader = null;
+        try {
+            getHttpURLConnection.setRequestMethod("GET");
+
+            setBasicHeaders(getHttpURLConnection);
+            setAdditionalAndDecoratedHeaders(getHttpURLConnection, additionalHeaders);
+
+            if (options.cacheControlHeadersEnabled()) {
```

Comment:
what happen if `options.targetCN()` is != `null` ? are you ignoring him? 

or if `options.hasCustomCN()`  is true?

Reply:
This is the same logic in the SplitHttpClientImpl, this option is checked on the Segment and Split Fetcher classes